Ghidra/Features/Decompiler/buildNatives.gradle

```diff
@@ -147,90 +176,113 @@ model {
+						}
+					} // end cpp
+				} // end sources decomp_dbg
+				binaries.all { b ->
+					if (b.toolChain in Gcc || b.toolChain in Clang) {
+						b.cppCompiler.args "-g"
+					}
+					else if (b.toolChain in VisualCpp) {
+						b.cppCompiler.args "/Zi"
+						b.cppCompiler.args "/FS"
+						b.linker.args "/DEBUG"
+					}
+				}
+			} // end decomp_dbg
+
+			sleigh_dbg(NativeExecutableSpec) {
+				targetPlatform "win_x86_64"
+				targetPlatform "linux_x86_64"
+				targetPlatform "linux_arm_64"
+				targetPlatform "mac_x86_64"
+				targetPlatform "mac_arm_64"
+				sources {
+					cpp {
+						source {
+							srcDir "src/decompile/cpp"
+							include common_files
+							include sleigh_files
+						}
+						exportedHeaders {
+							srcDir "src/decompile/cpp"
+						}
+					} // end cpp
+				} // end sources sleigh_dbg
+				binaries.all { b ->
+					if (b.toolChain in Gcc || b.toolChain in Clang) {
+						b.cppCompiler.args "-g"
+					}
+					else if (b.toolChain in VisualCpp) {
+						b.cppCompiler.args "/Zi"
+						b.cppCompiler.args "/FS"
+						b.linker.args "/DEBUG"
+					}
+				}
+			} // end sleigh_dbg
+		} // end flavors.debug
 	}  // end components
 	
-	binaries {
-		all{ b ->
-			if (b.toolChain in Gcc) {
-				b.cppCompiler.args "-std=c++11"
-				b.cppCompiler.args "-Wall"
-				b.cppCompiler.args "-O2"			// for DEBUG, comment this line out
-				// b.cppCompiler.args "-g"			// for DEBUG, uncomment	this line
-				b.cppCompiler.args "-Wno-sign-compare"
-				if (b.targetPlatform.operatingSystem.linux) {
-//					b.linker.args "-static"
-					b.cppCompiler.define "LINUX"
-					b.cppCompiler.define "_LINUX"
-				}
-			}
-		 	else if (b.toolChain in VisualCpp) {
-				b.cppCompiler.args "/EHsc"
-				b.cppCompiler.define "_SECURE_SCL=0"
-				b.cppCompiler.define "_HAS_ITERATOR_DEBUGGING=0"
-				// b.cppCompiler.args "/Zi"		// for DEBUG, uncomment this line
-				// b.cppCompiler.args "/FS"		// for DEBUG, uncomment this line
-				// b.linker.args "/DEBUG"		// for DEBUG, uncomment this line
-				if (b.targetPlatform.operatingSystem.windows) {
-					b.cppCompiler.define "WINDOWS"
-					b.cppCompiler.define "_WINDOWS"
-					b.cppCompiler.define "WIN32"
-					b.cppCompiler.define "_WIN32"
-					if (b.targetPlatform.name == "win_x86_64") {
-						b.cppCompiler.define "WIN64"
-						b.cppCompiler.define "_WIN64"
-					}	
-				}	
+	binaries.all { b ->
+		if (b.toolChain in Gcc || b.toolChain in Clang) {
+			b.cppCompiler.args "-std=c++11"
+			b.cppCompiler.args "-Wall"
+			b.cppCompiler.args "-Wno-sign-compare"
+			if (b.targetPlatform.operatingSystem.linux) {
+				b.cppCompiler.define "LINUX"
+				b.cppCompiler.define "_LINUX"
 			}
-			else if (b.toolChain in Clang) {
-				b.cppCompiler.args "-std=c++11"
-				b.cppCompiler.args "-Wall"
-				b.cppCompiler.args "-O2"			// for DEBUG, comment this line out
-				// b.cppCompiler.args "-g"			// for DEBUG, uncomment	this line
-				b.cppCompiler.args "-Wno-sign-compare"
-				b.cppCompiler.args "-w"
```

Comment:
It didn't make sense: With line 225 first all warnings should printed for Clang. Here all warnings should then be suppressed. I assumed -Wall is preferred and so I could merge flags for Gcc and Clang :)